remote-trait-object/src/service/export_import.rs

```diff
@@ -14,77 +14,68 @@
 // You should have received a copy of the GNU Affero General Public License
 // along with this program.  If not, see <https://www.gnu.org/licenses/>.
 
+use super::Dispatch;
 use super::*;
 use parking_lot::RwLock;
+use std::sync::Arc;
 
 // These traits are associated with some specific service trait.
 // These tratis will be implement by `dyn ServiceTrait` where `T = dyn ServiceTrait` as well.
 // Macro will implement this trait with the target(expanding) service trait.
 
-pub trait ExportServiceBox<T: ?Sized + Service> {
-    fn export(port: Weak<dyn Port>, object: Box<T>) -> HandleToExchange;
+/// Unused T is for avoiding violation of the orphan rule
+/// P will be local type for the crate, and that makes it possible to
+/// ```ignore
+/// impl ToDispatcher<dyn MyService> for Box<dyn MyService>
+/// ```
+pub trait ToDispatcher<T: ?Sized + Service> {
+    fn to_dispatcher(self) -> Arc<dyn Dispatch>;
 }
 
-pub trait ExportServiceArc<T: ?Sized + Service> {
-    fn export(port: Weak<dyn Port>, object: Arc<T>) -> HandleToExchange;
-}
-
-pub trait ExportServiceRwLock<T: ?Sized + Service> {
-    fn export(port: Weak<dyn Port>, object: Arc<RwLock<T>>) -> HandleToExchange;
-}
-
-pub trait ImportServiceBox<T: ?Sized + Service> {
-    fn import(port: Weak<dyn Port>, handle: HandleToExchange) -> Box<T>;
-}
-
-pub trait ImportServiceArc<T: ?Sized + Service> {
-    fn import(port: Weak<dyn Port>, handle: HandleToExchange) -> Arc<T>;
-}
-
-pub trait ImportServiceRwLock<T: ?Sized + Service> {
-    fn import(port: Weak<dyn Port>, handle: HandleToExchange) -> Arc<RwLock<T>>;
+/// Unused T is for avoiding violation of the orphan rule, like `ToDispatcher`
+pub trait ToRemote<T: ?Sized + Service> {
+    fn to_remote(port: Weak<dyn Port>, handle: HandleToExchange) -> Self;
```

Comment:
I don't understand why we need `ToDispatcher` and `ToRemote`. Could you explain it to me?